tests/test_page.py

```diff
@@ -10,23 +10,48 @@
 
 
 def test_base_url(base_url, page):
-    assert base_url == page.url
+    assert base_url == page.canonical_url
 
 
-def test_url_keywords(base_url, selenium):
+def test_canonical_url_absolute(base_url, selenium):
+    url_template = 'https://www.test.com/'
+
+    class MyPage(Page):
+        URL_TEMPLATE = url_template
+    page = MyPage(selenium, base_url)
+    assert url_template == page.canonical_url
+
+
+def test_canonical_url_empty(selenium):
+    page = Page(selenium)
+    assert page.canonical_url == ''
```

Comment:
Its an empty string because of my use of `urljoin`. I simply return the result of that, which is an empty string if both `self.base_url` and `self.URL_TEMPLATE` are empty. I could change that code to return `None` if the result of `urljoin` is empty. Would you prefer that?